### PR TITLE
Fix genemap2 parsing

### LINKED ISSUE #4162: Change in header of genemap2.txt is likely having effects on the resources parsing
This is the latest header
`# Chromosome	Genomic Position Start	Genomic Position End	Cyto Location	Computed Cyto Location	MIM Number	Gene/Locus And Other Related Symbols	Gene Name	Approved Gene Symbol	Entrez Gene ID	Ensembl Gene ID	Comments	Phenotypes	Mouse Gene Symbol/ID`

These are the fields of the header we used to parse (taken from demo files)
`# Chromosome	Genomic Position Start	Genomic Position End	Cyto Location	Computed Cyto Location	MIM Number	Gene Symbols	Gene Name	Approved Symbol	Entrez Gene ID	Ensembl Gene ID	Comments	Phenotypes	Mouse Gene Symbol/ID`

### PR DESCRIPTION
**Pull request by northwestwitch**
**Simpler to review and test after #4156** 

FIx #4162 -> Parse both old and new format of genemap2.txt files

<details>
<summary>Testing on cg-vm1 server (Clinical Genomics Stockholm)</summary>

**Prepare for testing**
1. Make sure the PR is pushed and available on [Docker Hub](https://hub.docker.com/repository/docker/clinicalgenomics/scout-server-stage)
1. Fist book your testing time using the Pax software available at [https://pax.scilifelab.se/](https://pax.scilifelab.se). The resource you are going to call dibs on is `scout-stage` and the server is `cg-vm1`.
1. `ssh <USER.NAME>@cg-vm1.scilifelab.se`
1. `sudo -iu hiseq.clinical`
1. `ssh localhost`
1. (optional) Find out which scout branch is currently deployed on cg-vm1: `podman ps`
1. Stop the service with current deployed branch: `systemctl --user stop scout.target`
1. Start the scout service with the branch to test: `systemctl --user start scout@<this_branch>`
1. Make sure the branch is deployed: `systemctl --user status scout.target`
1. After testing is done, repeat procedure at [https://pax.scilifelab.se/](https://pax.scilifelab.se), which will release the allocated resource (`scout-stage`) to be used for testing by other users.
</details>

<details>
<summary>Testing on hasta server (Clinical Genomics Stockholm)</summary>

**Prepare for testing**
1. `ssh <USER.NAME>@hasta.scilifelab.se`
1. Book your testing time using the Pax software. `us; paxa -u <user> -s hasta -r scout-stage`. You can also use the WSGI Pax app available at [https://pax.scilifelab.se/](https://pax.scilifelab.se).
1. (optional) Find out which scout branch is currently deployed on cg-vm1: `conda activate S_scout; pip freeze | grep scout-browser`
1. Deploy the branch to test: `bash /home/proj/production/servers/resources/hasta.scilifelab.se/update-tool-stage.sh -e S_scout -t scout -b <this_branch>`
1. Make sure the branch is deployed: `us; scout --version`
1. After testing is done, repeat the `paxa` procedure, which will release the allocated resource (`scout-stage`) to be used for testing by other users.
</details>

**How to test**:
1. how to test it, possibly with real cases/data

**Expected outcome**:
The functionality should be working
Take a screenshot and attach or copy/paste the output.

**Review:**
- [x] code approved by DN
- [x] tests executed by CR

### CODE REVIEW

File: scout/parse/omim.py [hunk 1]
```diff
@@ -146,13 +146,19 @@ def parse_genemap2(lines):
         parsed_entry["raw"] = line
 
         # This is the approved symbol for the entry
-        hgnc_symbol = parsed_entry.get("Approved Symbol")
+        approved_gene_symbol_col = parsed_entry.get(
+            "Approved Symbol", parsed_entry.get("Approved Gene Symbol")
+        )
+        hgnc_symbol = (
+            approved_gene_symbol_col.strip() if approved_gene_symbol_col not in ["", None] else None
```
Comment (dnil) [inline scout/parse/omim.py:153]: 👍

File: scout/parse/omim.py [hunk 2]
```diff
@@ -146,13 +146,19 @@ def parse_genemap2(lines):
...
+        )
 
         # If no approved symbol could be found choose the first of
         # the gene symbols
-        gene_symbols = []
-        if parsed_entry.get("Gene Symbols"):
-            gene_symbols = [symbol.strip() for symbol in parsed_entry["Gene Symbols"].split(",")]
+        gene_symbols_col = parsed_entry.get(
+            "Gene Symbols", parsed_entry.get("Gene/Locus And Other Related Symbols")
```
Comment (dnil) [inline scout/parse/omim.py:159]: 👍

File: scout/parse/omim.py [hunk 3]
```diff
@@ -146,13 +146,19 @@ def parse_genemap2(lines):
         parsed_entry["raw"] = line
 
         # This is the approved symbol for the entry
-        hgnc_symbol = parsed_entry.get("Approved Symbol")
+        approved_gene_symbol_col = parsed_entry.get(
+            "Approved Symbol", parsed_entry.get("Approved Gene Symbol")
```
Comment (dnil) [inline scout/parse/omim.py:150]: We should introduce a set of headings to expect from the input data.

**Review by dnil — APPROVED** [review]
It would not be strange to introduce a set of approved of headings to check the header names against before flushing them into the system. But this parsing has evidently been dysfunctional for a while, and thankfully HPO seems to have covered well, so we can also just make a new PR with that later.

File: scout/parse/omim.py (hunk 3 above)
Comment (northwestwitch) [reply scout/parse/omim.py:150]: I agree!

**Pull request merged**